docs/contributors/components.md

```diff
@@ -0,0 +1,40 @@
+# Storybook & Components
+
+This repo includes [Storybook](https://storybook.js.org) tooling so we can test and develop components in isolation.
+
+The storybook is automatically built and published to [GitHub pages](https://woocommerce.github.io/woocommerce-gutenberg-products-block/) on every push to the main branch. See [travis.yml](https://github.com/woocommerce/woocommerce-gutenberg-products-block/blob/main/.travis.yml) for details.
+
+https://woocommerce.github.io/woocommerce-gutenberg-products-block/
+
+## Where are our components?
+We have components in a few folders, for different contexts.
+
+- [`assets/js/base/components`](https://github.com/woocommerce/woocommerce-gutenberg-products-block/tree/main/assets/js/base/components)
+- [`assets/js/components`](https://github.com/woocommerce/woocommerce-gutenberg-products-block/tree/main/assets/js/components)
+- [`assets/js/icons`](https://github.com/woocommerce/woocommerce-gutenberg-products-block/tree/main/assets/js/icons)
+
+__`assets/js/base/components`__ are used in front-end code (as well as editor & admin). 
+These components are typically general purpose (and the largest collection). 
+These components help us build consistent interfaces across the front end (shopper) experience and elsewhere. 
+In time, some of these components may move to a general purpose lib, such as [`@wordpress/components` in Gutenberg](https://github.com/WordPress/gutenberg/tree/master/packages/components).
+
+__`assets/js/components`__ are used in the editor UI for our blocks. 
+These components are typically more specialised to WooCommerce Blocks. 
+They allow us to build a consistent and powerful UI for merchants for authoring content relating to Woo data - e.g. selecting products or product attributes.
+
```

Comment:
I'm not 100% sure about the distinction saying `assets/js/base/components` have a more general purpose while `assets/js/components` are more specialized to WC Blocks.

I think some components under `assets/js/components` might be useful in other projects outside WC Blocks (ProductsControl, PageSelector, etc.). In the other hand, some components under `assets/js/base/components` are clearly unique to WC Blocks like some C&C components.

In my opinion, the unique distinction between those two directories is that one is used for editor components and the other want for components that can be used either in the editor or the frontend.

Reply:
That's true - it's not easy to characterise the different groups, other than editor/non-editor (though there are probably some counterexamples in `base/components`).

By the way, when I say "general purpose" I mean things like radio buttons, checkboxes etc - not specific to woo at all, and even potentially useful outside of a WordPress context. If components are closer to general purpose, they're much more likely to graduate to another repo. 

Many of our components are at least partially specialised - to woo blocks, GB editor, or woocommerce data. I think it's quite tricky/subtle to find good candidates for reuse here. 

Let's see what feedback we get from the rest of the team on the meaning / intention of the two folders, then update the wording.

Reply:
Interesting that we're coming back around to `editor-components`!